### PR TITLE
Add `convert_to_dataset_fold`

### PR DESCRIPTION
**Pull request by Innixma**
Issue #, if available:

Description of changes:

- Add `convert_to_dataset_fold` to separate dataset and fold into separate columns if they were combined earlier.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/autogluon/bench/eval/evaluation/benchmark_evaluator.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def _load_results(
             results_raw = paths
         with warnings.catch_warnings():
             warnings.filterwarnings("ignore")
-            results_raw[TIME_INFER_S][results_raw[TIME_INFER_S] == 0] = 0.001
+            results_raw.loc[results_raw[TIME_INFER_S] == 0, TIME_INFER_S] = 0.001
```
Comment (Innixma) [inline src/autogluon/bench/eval/evaluation/benchmark_evaluator.py:145]: Resolves deprecation warning

File: src/autogluon/bench/eval/scripts/run_evaluation_openml.py [hunk 2]
```diff
@@ -106,12 +107,12 @@ def evaluate_amlb_results(
 # TODO: Rename to a more description function, or convert to a class
 def evaluate(
     *,
-    frameworks_run: List[str],
-    paths: List[str],
+    paths: List[str] | pd.DataFrame,
+    frameworks_run: List[str] | None = None,
```
Comment (Innixma) [inline src/autogluon/bench/eval/scripts/run_evaluation_openml.py:111]: Accept None as default to use all frameworks

File: src/autogluon/bench/eval/evaluation/benchmark_evaluator.py [hunk 3]
```diff
@@ -297,6 +297,8 @@ def filter_datasets(
 
         if max_rows is not None:
             task_metadata = task_metadata[task_metadata["NumberOfInstances"] <= max_rows]
+        if min_rows is not None:
+            task_metadata = task_metadata[task_metadata["NumberOfInstances"] >= min_rows]
```
Comment (Innixma) [inline src/autogluon/bench/eval/evaluation/benchmark_evaluator.py:301]: Added missing `min_rows` logic

**Review by suzhoum — APPROVED** [review]
LGTM!

**Pull request merged**